control-core/src/actors/mitsubishi_inverter_rs485.rs

```diff
@@ -0,0 +1,167 @@
+        Self {
+            received_response,
+            serial_interface,
+            init_done: false,
+            last_ts: Instant::now(),
+            last_op: Operation::None,
+        }
+    }
+}
+
+pub fn response_is_exception(response: ModbusResponse) -> bool {
+    let code: u8 = response.function_code.into();
+    return (code & 0b10000000) > 0; // 0x80 is set when an exception happens
+}
+
+pub fn response_function_code_is_exception(function_code: u8) -> bool {
+    return (function_code & 0b10000000) > 0; // 0x80 is set when an exception happens
+}
+
+enum RequestType {
+    OperationCommand,
+    /// Monitoring, Operation (start,stop etc) command, frequency setting (RAM), less than 12 milliseconds timeout
+    ReadWrite,
+    /// Parameter Read/Write and Frequency (EEPROM), Less than 30 milliseconds timeout
+    ParamClear,
+    /// Less than 5 seconds timeout
+    Reset,
+}
+
+impl RequestType {
+    fn timeout_milliseconds(self) -> u64 {
+        match self {
+            RequestType::OperationCommand => 12,
+            RequestType::ReadWrite => 30,
+            RequestType::ParamClear => 5000,
+            RequestType::Reset => 0,
+        }
+    }
+
+    fn timeout_nanoseconds(self) -> u64 {
+        self.timeout_milliseconds() * 1000000
+    }
+}
+
+enum MitsubishiModbusExceptionCode {
+    IllegalFunction,
+    IllegalDataAddress,
+    IllegalDataValue,
+    None,
+}
+
+impl MitsubishiModbusExceptionCode {
+    fn display(self) -> String {
+        match self {
+            MitsubishiModbusExceptionCode::IllegalFunction => "Illegal Function".to_string(),
+            MitsubishiModbusExceptionCode::IllegalDataAddress => "Illegal Data Address".to_string(),
+            MitsubishiModbusExceptionCode::IllegalDataValue => "Illegal Data Value".to_string(),
+            MitsubishiModbusExceptionCode::None => "No Exceptions".to_string(),
+        }
+    }
+}
+
+impl From<MitsubishiModbusExceptionCode> for u8 {
+    fn from(value: MitsubishiModbusExceptionCode) -> Self {
+        match value {
+            MitsubishiModbusExceptionCode::None => 0,
+            MitsubishiModbusExceptionCode::IllegalFunction => 1,
+            MitsubishiModbusExceptionCode::IllegalDataAddress => 2,
+            MitsubishiModbusExceptionCode::IllegalDataValue => 3,
+        }
+    }
+}
+
+impl From<u8> for MitsubishiModbusExceptionCode {
+    fn from(value: u8) -> Self {
+        match value {
+            1 => MitsubishiModbusExceptionCode::IllegalFunction,
+            2 => MitsubishiModbusExceptionCode::IllegalDataAddress,
+            3 => MitsubishiModbusExceptionCode::IllegalDataValue,
+            _ => MitsubishiModbusExceptionCode::None,
+        }
+    }
+}
+
+impl Actor for MitsubishiInverterRS485Actor {
+    fn act(&mut self, _now_ts: Instant) -> Pin<Box<dyn Future<Output = ()> + Send + '_>> {
+        Box::pin(async move {
+            let elapsed: Duration = self.last_ts.elapsed();
```

Comment:
use `duration_since` with `now_ts` because elapsed is internally creating a new timestamp. It's not critical or _bad_ just inconsistent with later setting `last_ts` to `now_ts`.